monkey/infection_monkey/telemetry/tests/test_base_telem_classes.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+
+from infection_monkey.exploit.wmiexec import WmiExploiter
+from infection_monkey.model.host import VictimHost
+from infection_monkey.post_breach.actions.schedule_jobs import ScheduleJobs
+from infection_monkey.telemetry.exploit_telem import ExploitTelem
+from infection_monkey.telemetry.post_breach_telem import PostBreachTelem
+from infection_monkey.telemetry.scan_telem import ScanTelem
+from infection_monkey.telemetry.state_telem import StateTelem
+from infection_monkey.telemetry.system_info_telem import SystemInfoTelem
+from infection_monkey.telemetry.trace_telem import TraceTelem
+from infection_monkey.telemetry.tunnel_telem import TunnelTelem
+
+
+DOMAIN_NAME = 'domain-name'
+HOSTNAME = 'hostname'
+IP = '0.0.0.0'
+IS_DONE = True
+MSG = 'message'
+RESULT = False
+SYSTEM_INFO = {}
+VERSION = 'version'
+HOST = VictimHost(IP, DOMAIN_NAME)
+EXPLOITER = WmiExploiter(HOST)
+PBA = ScheduleJobs()
+
+
+@pytest.fixture
+def exploit_telem_test_instance():
+    return ExploitTelem(EXPLOITER, RESULT)
+
+
+def test_exploit_telem_category(exploit_telem_test_instance):
+    assert exploit_telem_test_instance.telem_category == 'exploit'
+
+
+def test_exploit_telem_get_data(exploit_telem_test_instance):
+    actual_data = exploit_telem_test_instance.get_data()
+    expected_data = {'result': RESULT,
+                     'machine': HOST.as_dict(),
+                     'exploiter': EXPLOITER.__class__.__name__,
+                     'info': EXPLOITER.exploit_info,
+                     'attempts': EXPLOITER.exploit_attempts}
+    assert actual_data == expected_data
+
+
+@pytest.fixture
+def post_breach_telem_test_instance(mocker):
```

Comment:
Prefer [monkeypatch](https://docs.pytest.org/en/stable/monkeypatch.html) which is already part of pytest

Reply:
Resolved in 4efdeeacc